gcc/rust/backend/rust-compile-pattern.cc

```diff
@@ -329,5 +335,134 @@ CompilePatternBindings::visit (HIR::StructPattern &pattern)
+						       rval_locus);
+      auto s = ctx->get_backend ()->init_statement (fnctx.fndecl, var,
+						    unit_type_init_expr);
+      ctx->add_statement (s);
+    }
+  else
+    {
+      auto s
+	= ctx->get_backend ()->init_statement (fnctx.fndecl, var, init_expr);
+      ctx->add_statement (s);
+    }
+}
+
+void
+CompilePatternLet::visit (HIR::GroupedPattern &pattern)
+{
+  CompilePatternLet::Compile (&*pattern.get_item (), init_expr, ty, rval_locus,
+			      ctx);
+}
+
+void
+CompilePatternLet::visit (HIR::WildcardPattern &pattern)
+{
+  Bvariable *var = nullptr;
+  if (!ctx->lookup_var_decl (pattern.get_pattern_mappings ().get_hirid (),
+			     &var))
+    {
+      // FIXME this should be an assertion instead and use error mark node
+      rust_fatal_error (pattern.get_locus (),
+			"failed to lookup compiled variable declaration");
+      return;
+    }
+
+  auto fnctx = ctx->peek_fn ();
+  if (ty->is_unit ())
+    {
+      ctx->add_statement (init_expr);
+
+      tree stmt_type = TyTyResolveCompile::compile (ctx, ty);
+
+      auto unit_type_init_expr
+	= ctx->get_backend ()->constructor_expression (stmt_type, false, {}, -1,
+						       rval_locus);
+      auto s = ctx->get_backend ()->init_statement (fnctx.fndecl, var,
+						    unit_type_init_expr);
+      ctx->add_statement (s);
+    }
+  else
+    {
+      auto s
+	= ctx->get_backend ()->init_statement (fnctx.fndecl, var, init_expr);
+      ctx->add_statement (s);
+    }
+}
+
+void
+CompilePatternLet::visit (HIR::TuplePattern &pattern)
+{
+  rust_assert (!ty->is_unit ());
+  rust_assert (pattern.has_tuple_pattern_items ());
+  switch (pattern.get_items ()->get_pattern_type ())
+    {
+    default:
+      gcc_unreachable ();
+    case HIR::TuplePatternItems::TuplePatternItemType::RANGED:
+      rust_fatal_error (pattern.get_locus (),
+			"ranged tuple pattern in let statement");
+      return;
+    case HIR::TuplePatternItems::TuplePatternItemType::MULTIPLE:
+      break;
+    }
+
+  tree tuple_type = TyTyResolveCompile::compile (ctx, ty);
+  tree init_stmt;
+  Bvariable *tmp_var
+    = ctx->get_backend ()->temporary_variable (ctx->peek_fn ().fndecl,
+					       NULL_TREE, tuple_type, init_expr,
+					       false, pattern.get_locus (),
+					       &init_stmt);
+  tree access_expr
+    = ctx->get_backend ()->var_expression (tmp_var, pattern.get_locus ());
+  ctx->add_statement (init_stmt);
+
+  size_t tuple_idx = 0;
+  auto &items
+    = static_cast<HIR::TuplePatternItemsMultiple &> (*pattern.get_items ());
+  for (auto &sub : items.get_patterns ())
+    {
+      TyTy::BaseType *ty_sub = nullptr;
+      HirId pattern_id = pattern.get_pattern_mappings ().get_hirid ();
+      bool ok = ctx->get_tyctx ()->lookup_type (pattern_id, &ty_sub);
+      rust_assert (ok);
+
+      tree sub_init
+	= ctx->get_backend ()->struct_field_expression (access_expr, tuple_idx,
+							sub->get_locus ());
+      CompilePatternLet::Compile (&*sub, sub_init, ty_sub, rval_locus, ctx);
```

Comment:
```suggestion
      CompilePatternLet::Compile (sub.get (), sub_init, ty_sub, rval_locus, ctx);
```
Nit, but isn't this effectively the same?

Reply:
Yeah. Do I do anything different with Signed-off-by for applying this?

Reply:
No, it's fine. You can add a `Co-authored-by: name <email>` line if you'd like, but for reviews you don't need to :)